rasa/server.py

```diff
@@ -625,7 +684,7 @@ async def execute_action(request: Request, conversation_id: Text):
 
         try:
             async with app.agent.lock_store.lock(conversation_id):
-                tracker = await get_tracker(
+                tracker = await get_tracker_with_session_start(
```

Comment:
This collides with the [API spec](https://rasa.com/docs/rasa/pages/http-api#operation/predictConversationAction) which states

> The state of the tracker is not modified.

I think it makes sense to keep the validation of an empty tracker for this (and to not modify the sessions)

Reply:
no hang on, the function was just badly named before:

https://github.com/RasaHQ/rasa/pull/6979/files#diff-394c9a945911d29e59b1eae237cdf29d7bc1276940b6dfc2cd95dd5a51117143L230

it was still executing the session start

Reply:
I guess fixing that should happen in a separate PR

Reply:
I think it would be great to tackle it right away as we're otherwise never gonna tackle it, but the PR already has a decent size. Can you please create an issue? I think it would be good for someone who is getting started at Rasa.

Reply:
will handle this in a separate PR: https://github.com/RasaHQ/rasa/pull/7020